src/snowflake/cli/api/project/schemas/entities/common.py

```diff
@@ -59,6 +61,25 @@ def get_type(cls) -> str:
         return cls.model_fields["type"].annotation.__args__[0]
 
     meta: Optional[MetaField] = Field(title="Meta fields", default=None)
+    identifier: Optional[Union[Identifier | str]] = Field(
+        title="Entity identifier", default=None
+    )
+    key: str | None = None
```

Comment:
Is this just internal for convenience to have the key on the model? Or can users specify the key here as well?

Reply:
Changed that to `PrivateAttr` so users cannot set it and it's not part of model.